src/app/static/src/app/store/load/actions.ts

```diff
@@ -144,14 +144,17 @@ const getRehydrateResult = async (context: ActionContext<LoadState, RootState>)
         const {files, savedState} = constructRehydrateProjectState(rootState, response.data)
 
         if (!rootGetters.isGuest) {
-            await (dispatch("projects/createProject",
+            await dispatch("projects/createProject",
                 {
                     name: state.projectName,
                     isUploaded: true
-                }, {root: true}));
+                }, {root: true});
 
-            savedState.projects.currentProject = rootState.projects.currentProject
-            savedState.projects.currentVersion = rootState.projects.currentVersion
+            const project = {
+                currentProject: rootState.projects.currentProject,
+                currentVersion: rootState.projects.currentVersion
+            }
+            Object.assign(savedState.projects, project)
```

Comment:
Isn't this the wrong way round? It's `Object.assign(target, source)` so I was expecting you to do something like. `Object.assign(rootState, {...emptyState(), ...savedState})` having first constructed an empty root state. (And the problem isn't just about the  projects state, it's all of it.) Remember, the probem is that the current root state gets constantly saved to local storage, so that's overwriting your savedState.